src/System Application/App/Azure AD Plan/src/AzureADPlanImpl.Codeunit.al

```diff
@@ -356,6 +340,31 @@ codeunit 9018 "Azure AD Plan Impl."
         end;
     end;
 
+    [NonDebuggable]
+    local procedure ShouldCheckMixedPlans(): Boolean
+    var
+        Company: Record Company;
+        EnvironmentInformation: Codeunit "Environment Information";
+    begin
+        if not EnvironmentInformation.IsSaaS() then
+            exit(false);
+
+        if not GuiAllowed() then
```

Comment:
Why should it not check if Gui is not allowed?

Reply:
Indeed. This should work for API scenarios too.

Reply:
We've never checked mixed plans when Gui is not allowed, so this would be a new enforcement.

Might be quite a lot of overhead to add to API scenarios?

Reply:
If you don't check it outside of GUIallowed sessions, then essentials can technically log into premium experience tier companies.

Reply:
Resolving comment as changing the design here would require proper testing and insights into the performance impact of that change.

Safest option is to keep the same design and create a follow up work item to look into this if relevant.